src/QMCWaveFunctions/lcao/LCAOrbitalBuilder.cpp

```diff
@@ -904,6 +904,13 @@ void LCAOrbitalBuilder::EvalPeriodicImagePhaseFactors(PosType SuperTwist,
 void LCAOrbitalBuilder::EvalPeriodicImagePhaseFactors(PosType SuperTwist,
                                                       std::vector<std::complex<RealType>>& LocPeriodicImagePhaseFactors)
 {
+  // For open boundary conditions
+  if (PBCImages[0] == 0 && PBCImages[1] == 0 && PBCImages[2] == 0) {
+    LocPeriodicImagePhaseFactors.resize(1);
+    LocPeriodicImagePhaseFactors[0] = 1.0;
+    return;
+  }
+
```

Comment:
I feel better not using early return.
How about protect hdf5 read with
```
if  (!(PBCImages[0] == 0 && PBCImages[1] == 0 && PBCImages[2] == 0))
```

Reply:
I'm concerned that avoiding the early return is going to complicate the logic and make it harder to read.

Reply:
To me adding early return creates two fully divergent code paths and complicates the logic. If we don't need to distinguish no-PBC and PBC, it is better not to do it. However, we need to add proper comment saying PBC case needs to read lattice vectors.

Reply:
I will leave it to others to express their preferences.

Reply:
I worked through the TransX logic in the loop below the HDF file read and verified it gives the same result, and I found a way to add the extra logic.   I will update with the early return removed as soon as the local build works.